docs/contributing/index.md

```diff
@@ -228,3 +228,46 @@ They stand out better from the main text, and can be collapsed by default if nee
     This note is collapsed, because it uses `???`.
     
 If an admonition is collapsed by default, it should have a title.
+
+### Code blocks
+
+Use [code blocks](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/) when you want to display monospace text in a programming language, terminal output, configuration files etc.
+The documentation uses [pygments](https://pygments.org) for highlighting.
+See [list of available lexers](https://pygments.org/docs/lexers/#) for the languages that you can use for code blocks.
+
+Use [`console`](https://pygments.org/docs/lexers/#pygments.lexers.shell.BashSessionLexer) for interactive sessions with prompt-output pairs:
+````markdown
+```console title="Hello, world!"
+$ echo "Hello, world!"
+Hello, world!
+```
+````
+
+The above becomes:
+
+```console title="Hello, world!"
+$ echo "Hello, world!"
+Hello, world!
+```
+
+!!! warning
+    `terminal` is not a valid lexer, but MkDocs or pygments will not warn about using it as a language.
+    The text will be rendered without highlighting.
+
+Note the use of `title=...`, which will give the code block a heading.
+
+!!! tip
+    Include a title whenever possible to describe what the code block does or is.
+
+If you want to display commands without that can easily be copied, use `bash` as the language.
+````markdown
+```bash title="Hello, world!"
+echo "Hello, world!"
+```
+````
+
+The above becomes:
+
+```bash title="Hello, world!"
+echo "Hello, world!"
+```
```

Comment:
From these docs and updates on other pages, it looks like you lean towards using `$` for the prompt in shell code blocks instead of `$`.

How about making this a formal policy, and documenting it here?

Reply:
Yeah, sorry, it was actually not so much a preference, but a must. `>` doesn't trigger correct highlighting, whereas pygments recognizes `$` as the actual prompt.

I'll make it clear that `$` is actually a requirement, not just a preference.

Reply:
https://github.com/eth-cscs/cscs-docs/pull/74/commits/12475300572e16f29aac53be0a9653cf3ab652a5